src/shims/env.rs

```diff
@@ -29,6 +30,7 @@ impl EnvVars {
                 }
             }
         }
+        ecx.update_environ().unwrap();
```

Comment:
Please propagate the error outwards. We are unwrapping [here](https://github.com/rust-lang/miri/blob/e6a0c60a0feba65b2ced98b3eae4a2406a7d635e/src/eval.rs#L199) and that should be the only place.